### PR TITLE
[MCT-2077] TCP and DNS checks

### PR DESCRIPTION
**Pull request by BoRuDar**

### CODE REVIEW

File: pingdom/param/constants.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package param
+
```
Comment (rafaykh90) [inline pingdom/param/constants.go:2]: There are a few params missing e.g. `stringtoexpect`, `stringtosend` for TCP checks, and few others for HTTP checks as well. Is it by choice?

File: pingdom/checkTypes.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
+package pingdom
+
+import (
+	"github.com/nordcloud/go-pingdom"
+)
+
+func convertIntoHTTPCheck(c allCheckParams) *pingdom.HTTP {
```
Comment (rafaykh90) [inline pingdom/checkTypes.go:7]: Using it like `func (c allCheckParams) ToHTTPCheck() *pingdom.HTTP`  and then calling the method like `checkParams.ToHTTPCheck()` seems a bit cleaner way to me. I am not an expert in GO so just wondering is there some issue with such approach. 

Same goes for the rest of the conversion methods

File: pingdom/provider.go [hunk 3]
```diff
@@ -25,12 +28,11 @@ func Provider() terraform.ResourceProvider {
...
-	if err := mapstructure.Decode(configRaw, &config); err != nil {
-		return nil, err
+	pingdomAPIToken, ok := d.Get(param.APIToken).(string)
+	if !ok {
+		return nil, fmt.Errorf("API token for Pingom is NOT provided")
 	}
 
 	log.Println("[INFO] Initializing Pingdom client")
-	return config.Client()
+	return pingdom.New(pingdomAPIToken), nil
```
Comment (rafaykh90) [inline pingdom/provider.go:37]: I understand that this method is from the `go-pingdom` package, but the method name `New` seems a bit ambiguous. Should it be something like `GetClient` or `CreateClient`?

Comment (BoRuDar) [reply pingdom/provider.go:37]: It's kind of convention in Go to use `New` word in constructors.

File: pingdom/param/constants.go (hunk 1 above)
Comment (BoRuDar) [reply pingdom/param/constants.go:2]: Somehow it was missed. I'll re-check it. Thanks

File: pingdom/checkTypes.go (hunk 2 above)
Comment (BoRuDar) [reply pingdom/checkTypes.go:7]: Hm, it looks a bit better with your approach. I'll refactor this. Thanks

Comment (BoRuDar) [reply pingdom/checkTypes.go]: done

File: pingdom/param/constants.go (hunk 1 above)
Comment (BoRuDar) [reply pingdom/param/constants.go:2]: done

**Pull request merged**